Lesson16/Gokhan/script.js

```diff
@@ -0,0 +1,234 @@
+
+function checkLeapYear(year) {
+
+    if ((year % 4 === 0 && year % 100 !== 0) || year % 400 === 0) {
+        console.log(year + " is a leap year.");
+    } else {console.log(year + " is not a leap year.")}
+
+}
+    // I didn't do this myself (with AI)
+    checkLeapYear(2020);
+    checkLeapYear(1900);
+    checkLeapYear(2000);
+
+// 7. Write a function that takes in a temperature and checks if it's above or below freezing using if else statement.
+
+function temperatureChecks (freezing) {
+
+    if (freezing > 0) {
+        console.log("The temperature is above the freezing.");
+    } else if (freezing < 0) {
+        console.log("The temperature is below the freezing.");
+    }
+
+}
+
+    temperatureChecks(27);
+    temperatureChecks(-15);
+
+// 8. Write a function that takes in a person's height and weight and checks 
+//    if they are considered underweight, normal, overweight, or obese using if else statement.
+//    Body Mass Index (BMI) is a person’s weight in kilograms (or pounds) divided by the square of height in meters (or feet). 
+//    A high BMI can indicate high body fatness.
+//    Formula: weight (kg) / [height (m)]2
+//    Below 18.5	Underweight
+//    18.5 – 24.9	Healthy Weight
+//    25.0 – 29.9	Overweight
+//    30.0 and Above	Obesity
+
+function checkBMI(weight, height) {
+
+    let bmi = weight / (height * height);
+
+    if (bmi < 18.5) {
+        console.log("Underweight");
+    } else if (bmi >= 18.5 && bmi <= 24.9) {
+        console.log("Healthy Weight");
+    } else if (bmi >= 25.0 && bmi <= 29.9) {
+        console.log("Overweight");
+    } else {console.log("Above Obesity")}
+
+}
+
+    checkBMI(15, 1.6);
+    checkBMI(21, 1.5);
+    checkBMI(28, 1.8);
+    checkBMI(36, 1.2);
+
+// 9. Write a function that takes in a person's age and checks 
+//     if they are a child (age 0-12), teenager (age 13-19), adult (age 20-64), 
+//     or senior citizen (age 65 and above) using if else statement.
+
+function personAge(age) {
+
+    if (age >= 0 && age <= 12) {
+        console.log("You are child.");
+    } else if (age >= 13 && age <= 19) {
+        console.log("You are teenager.");
+    } else if (age >= 20 && age <= 64) {
+        console.log("You are adult.");
+    } else if (age >= 65) {
+        console.log("You are senior citizen.");
+    } 
+
+}
+
+    personAge(7);
+    personAge(15);
+    personAge(41);
+    personAge(77);
+
+
+// 10. Write a function that takes age as a age and whether they are a student (yes or no). 
+//     If the age is less than 18 and the user is a student, print "You are a student and not old enough to vote". 
+//     If the age is between 18 and 65 (inclusive) and the user is not a student, print "You are eligible to vote". 
+//     If the age is greater than 65, print "You are a senior citizen". 
+//     If the user is a student but their age is greater than or equal to 18, print "You are old enough to vote".
+
+function checkVotingEligibility(age, isStudent) {
```

Comment:
For values like 'yes'/'no', in programming it's common to use Boolean type value. So change function body like this:

```
if (age < 18 && isStudent) {
        console.log("You are a student and not old enough to vote");
    } else if (age >= 18 && age <= 65 && !isStudent) {
        console.log("You are eligible to vote");
    } else if (age > 65) {
        console.log("You are a senior citizen");
    } else if (isStudent && age >= 18) {
        console.log("You are old enough to vote");
    }
```

See more info about booleans here: https://www.classes.cs.uchicago.edu/archive/2021/spring/11111-1/happycoding/p5js/if-statements.html#:~:text=An%20if%20statement%20checks%20a,when%20that%20value%20is%20true%20.

When calling the function, you need to call it like this: `checkVotingEligibility(17, true);`

Reply:
It is Done!